### PR TITLE
feat: from_text format:json

### PR DESCRIPTION
**Pull request by aljazerzen**
- json format
- changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # PRQL Changelog
 
+## 0.4.2 — [unreleased]
+
+**Features**:
+
+- `from_text` function that supports JSON and CSV formats. (@aljazerzen, @snth)
```
Comment (snth) [inline CHANGELOG.md:7]: Thanks. @max-sixty also needs a mention as he actually wrote the original `from_csv` and I just did the renaming to `from_text`.

Comment (aljazerzen) [reply CHANGELOG.md:7]: True, my bad

Comment (max-sixty) [reply CHANGELOG.md:7]: I did very little here! And I left it hanging for ages so I should get a negative call-out

File: prql-compiler/src/semantic/transforms.rs [hunk 2]
```diff
@@ -847,29 +875,113 @@ impl AstFold for Flattener {
...
-        rows: rdr
-            .records()
-            .into_iter()
-            // This is messy rust, but I can't get it to resolve the Errors
-            // when it leads with `row_result?`. I'm sure it's possible...
-            .map(|row_result| {
-                row_result.map(|row| row.into_iter().map(|x| x.to_string()).collect())
+    // TODO: Can we dynamically get the types, like in pandas? We need to put
+    // quotes around strings and not around numbers.
+    // https://stackoverflow.com/questions/64369887/how-do-i-read-csv-data-without-knowing-the-structure-at-compile-time
```
Comment (max-sixty) [inline prql-compiler/src/semantic/transforms.rs:883]: I actually think this wasn't a problem in the end?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+  '''
+
+  from_text format:json '''{
+      "columns": ["a", "b", "c"],
+      "data": [
+          [1, "x", false],
+          [4, "y", null]
+      ]
+  }'''
+  ```
```
Comment (max-sixty) [inline CHANGELOG.md:27]: Thanks for adding the Changlog. It's also basically docs!

**Comment by max-sixty** [comment]
Awesome @aljazerzen !

File: prql-compiler/src/semantic/transforms.rs (hunk 2 above)
Comment (aljazerzen) [reply prql-compiler/src/semantic/transforms.rs:883]: I just used `Literal` as a cell type so I can contain basically anything PRQL supports.

Comment (max-sixty) [reply prql-compiler/src/semantic/transforms.rs:883]: Perfect!

**Pull request merged**